### DISCUSSION TITLE
Connecting drone for swarm sequence

### DISCUSSION THREAD

**Discussion by Bikerholic** [bitcraze/discussions#99] [Q&A]
Hi
    I have a more confusion in connecting a 2 or more crazyflie in a single crazy radio or a multiple crazyradio for the swarm sequence please tell how to connect and fly a swarm sequence thanks in advance

**Comment by knmcguire** [comment]
Have you looked or tried any of the swarm examples in the cflib? https://github.com/bitcraze/crazyflie-lib-python/tree/master/examples/swarm

You first have to give each crazyflie an unique URI, which you can do via these instructions: https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/#firmware-configuration

Another option is also to look at the Crazyswarm project that uses ROS for this: https://crazyswarm.readthedocs.io/en/latest/. We don't maintain that one ourselves but it has better swarm support as compressed messages and such.

**Reply by thebikerholic** [reply]
I am clear about gaving unique URI to the crazyflies after gaving how to connect it to cf client without any gamepad or external devices we are trying to fly autonoumesly.

**Reply by knmcguire** [reply]
The CFclient does not support swarms, as it can only connect to one. 

Either need to use those scripts with the crazyflie-python-library and change the URIs in the python scripts to those that your crazyflies have, or give it a go on crazyswarm

**Comment by thebikerholic** [comment]
THANKS FOR THE REPLY,
How to change the scripts in python library and how to use it i want a detailed explaination about it sorry for inconvinience please help me i am a beginer i invest lot of money in this.

**Reply by knmcguire** [reply]
You will need to change [these lines ](https://github.com/bitcraze/crazyflie-lib-python/blob/a8eca1f8267c277dd4e19e18d5344cf033126915/examples/swarm/swarmSequence.py#L55-L64)to to the uri's of your crazyflies.

The Crazyflie is a development platform, so it is necessary to get into the coding yourselve unfortunately. We can help with some of it but it requires some reading to do for yourselves.

If you are a complete beginner, please follow the cflib's stepbystep guides to get a feel for the library: https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/user-guides/sbs_connect_log_param/

These lectures are perhaps also handy for you: https://www.youtube.com/watch?v=6s8i-nhPjt0&t=2697s. There is non on swarms though but you can interpolate that knowledge to more.
 
And the links to crazyswarm also have plenty of tutorials to work on as well.

Good luck!
